ibm/resource_ibm_private_dns_glb_monitor.go

```diff
@@ -0,0 +1,464 @@
+	monitorretries := int64(d.Get(pdnsGlbMonitorRetries).(int))
+	monitortimeout := int64(d.Get(pdnsGlbMonitorTimeout).(int))
+	CreateMonitorOptions.SetName(monitorname)
+	CreateMonitorOptions.SetInterval(monitorinterval)
+	CreateMonitorOptions.SetRetries(monitorretries)
+	CreateMonitorOptions.SetTimeout(monitortimeout)
+	if monitordescription, ok := d.GetOk(pdnsGlbMonitorDescription); ok {
+		CreateMonitorOptions.SetDescription(monitordescription.(string))
+	}
+	if Mtype, ok := d.GetOk(pdnsGlbMonitorType); ok {
+		CreateMonitorOptions.SetType(Mtype.(string))
+	}
+	if Mport, ok := d.GetOk(pdnsGlbMonitorPort); ok {
+		CreateMonitorOptions.SetPort(int64(Mport.(int)))
+	}
+	if monitorpath, ok := d.GetOk(pdnsGlbMonitorPath); ok {
+		CreateMonitorOptions.SetPath((monitorpath).(string))
+	}
+	if monitorexpectedcodes, ok := d.GetOk(pdnsGlbMonitorExpectedCodes); ok {
+		CreateMonitorOptions.SetExpectedCodes((monitorexpectedcodes).(string))
+	}
+	if monitormethod, ok := d.GetOk(pdnsGlbMonitorMethod); ok {
+		CreateMonitorOptions.SetMethod((monitormethod).(string))
+	}
+	if monitorexpectedbody, ok := d.GetOk(pdnsGlbMonitorExpectedBody); ok {
+		CreateMonitorOptions.SetExpectedBody((monitorexpectedbody).(string))
+	}
+	if monitorheaders, ok := d.GetOk(pdnsGlbMonitorHeaders); ok {
+		expandedmonitorheaders, err := expandGLBMonitors(monitorheaders)
+		if err != nil {
+			return err
+		}
+		CreateMonitorOptions.SetHeadersVar(expandedmonitorheaders)
+	}
+	if monitorallowinsecure, ok := d.GetOk(pdnsGlbMonitorAllowInsecure); ok {
+		CreateMonitorOptions.SetAllowInsecure((monitorallowinsecure).(bool))
+	}
+
+	response, detail, err := sess.CreateMonitor(CreateMonitorOptions)
+	if err != nil {
+		return fmt.Errorf("Error creating pdns GLB monitor:%s\n%s", err, detail)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", instanceID, *response.ID))
+	return resourceIBMPrivateDNSGLBMonitorRead(d, meta)
+}
+
+func expandGLBMonitors(header interface{}) ([]dnssvcsv1.HealthcheckHeader, error) {
+	headers := header.(*schema.Set).List()
+	expandheaders := make([]dnssvcsv1.HealthcheckHeader, 0)
+	for _, v := range headers {
+		locationConfig := v.(map[string]interface{})
+		hname := locationConfig[pdnsGlbMonitorHeadersName].(string)
+		headers := expandStringList(locationConfig[pdnsGlbMonitorHeadersValue].([]interface{}))
+		headerItem := dnssvcsv1.HealthcheckHeader{
+			Name:  &hname,
+			Value: headers,
+		}
+		expandheaders = append(expandheaders, headerItem)
+	}
+	return expandheaders, nil
+}
+
+func resourceIBMPrivateDNSGLBMonitorRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+	idset := strings.Split(d.Id(), "/")
+
+	getMonitorOptions := sess.NewGetMonitorOptions(idset[0], idset[1])
+	response, detail, err := sess.GetMonitor(getMonitorOptions)
+	if err != nil {
+		return fmt.Errorf("Error fetching pdns GLB Monitor:%s\n%s", err, detail)
+	}
+	d.Set("id", response.ID)
```

Comment:
setting "id" is equivalent to d.SetId() in create method..
Please comment this line

Reply:
handled